src/utils/flashMintNotional/fmNotionalQuote.ts

```diff
@@ -0,0 +1,80 @@
+import { BigNumber, ethers } from 'ethers'
+
+import { Provider } from '@ethersproject/abstract-provider'
+import { Signer } from '@ethersproject/abstract-signer'
+import { Exchange, SwapData } from '@indexcoop/flash-mint-sdk'
+
+import { getFlashMintNotionalContract } from './fmNotionalContract'
+
+export interface FlashMintNotionalQuote {
+  indexTokenAmount: BigNumber
+  inputOutputTokenAmount: BigNumber
+  swapData: SwapData[]
+}
+
+type FilteredComponentsResponse = {
+  filteredComponents: string[]
+  filteredUnits: BigNumber[]
+}
+
+export const getFlashMintNotionalQuote = async (
+  isMinting: boolean,
+  fixedTokenAddress: string,
+  inputOutputTokenAddress: string,
+  amountIndexToken: BigNumber,
+  slippagePercent: number,
+  providerOrSigner: Signer | Provider | undefined
+): Promise<FlashMintNotionalQuote> => {
+  const contract = getFlashMintNotionalContract(providerOrSigner)
+  const issuanceModule = '0xa0a98EB7Af028BE00d04e46e1316808A62a8fd59'
+  const isDebtIssuance = true
+  const slippage = ethers.utils.parseEther(slippagePercent.toString())
+
+  const { filteredComponents, filteredUnits }: FilteredComponentsResponse =
+    isMinting
+      ? await contract.getFilteredComponentsIssuance(
+          fixedTokenAddress,
+          amountIndexToken,
+          issuanceModule,
+          isDebtIssuance,
+          slippage
+        )
+      : await contract.getFilteredComponentsRedemption(
+          fixedTokenAddress,
+          amountIndexToken,
+          issuanceModule,
+          isDebtIssuance,
+          slippage
+        )
+
+  // Most swap data can be static if input is DAI/USDC since it won't be picked up.
+  const swapData = isMinting
+    ? filteredComponents.map((component: string) => {
+        return {
+          // input token adress
+          path: [inputOutputTokenAddress, component],
+          fees: [3000],
+          pool: ethers.constants.AddressZero,
+          exchange: Exchange.UniV3,
+        }
+      })
+    : filteredComponents.map((component: string) => {
+        return {
+          // output token adress
+          path: [component, inputOutputTokenAddress],
+          fees: [3000],
+          pool: ethers.constants.AddressZero,
+          exchange: Exchange.UniV3,
+        }
+      })
+
+  const inputOutputTokenAmount = filteredUnits.reduce((prevVal, currentVal) => {
```

Comment:
Not actually sure about this, but you might have to factor in slippage into this calculation. (maybe just something to keep in mind if we start seeing revertions on this in the future)

Reply:
Isn't `getFilteredComponentsIssuance` factoring this in? Because I pass slippage there:

```
await contract.getFilteredComponentsIssuance(
          fixedTokenAddress,
          amountIndexToken,
          issuanceModule,
          isDebtIssuance,
          slippage
        )
```

If not, or if you think it's safer in general, I'd add slippage to the `inputOutputTokenAmount`.

Reply:
You're right it should 👍 . Probably fine like this then. Still if we encounter issues / reversions in the future I think slippage setting is always a good first suspect to look at.